### PR TITLE
feat: setData

### PR DESCRIPTION
**Pull request by lmiller1990**
Still not a big fan of this feature, but at least it was easy to implement 👍

### CODE REVIEW

File: src/vue-wrapper.ts [hunk 1]
```diff
@@ -78,6 +79,13 @@ export class VueWrapper implements WrapperAPI {
     return Array.from(results).map((x) => new DOMWrapper(x))
   }
 
+  setData(data: Record<string, any>) {
+    // lodash.merge merges by *reference* so this will update
+    // any existing data with the newly passed data.
+    merge(this.componentVM.$data, data)
```
Comment (dobromir-hristov) [inline src/vue-wrapper.ts:85]: Should we guard against setting data that is not in `$data` already? Not sure

Comment (lmiller1990) [reply src/vue-wrapper.ts:85]: Let's check how Vue 3 behaves. I wonder if the new reactivity API allows you to update with new properties? I think we should match whatever Vue 3 does.

**Comment by lmiller1990** [comment]
Should I re-base and merge this? I forgot why we left this open.

**Comment by afontcu** [comment]
> Should I re-base and merge this? I forgot why we left this open.

I think we said to wait until later alphas, and see if people asked about this particular feature. I won't be using it personally, but some concerns were raised in the RFC so we might just add it and keep it as a low priority feature.

**Comment by lmiller1990** [comment]
Dropped lodash dep, will need to write our own `merge` function before merging this.

**Comment by lmiller1990** [comment]
I will close this for now, we can implement if/when we need it.

**Comment by jackmu95** [comment]
We at [BootstrapVue](https://github.com/bootstrap-vue/bootstrap-vue) are using `wrapper.setData()` quite heavily.
We are currently in the process of creating a Vue 3 compatible version and it would be good to have that feature in VTU v2.

I'am also willing to test this with our current tests and give feedback.

**Comment by JessicaSachs** [comment]
Would you like help in refactoring to move away from setData? I see 11 code
usages in bootstrap Vue (
https://github.com/bootstrap-vue/bootstrap-vue/search?q=setData&type=
) and I feel comfortable submitting a PR to migrate off of it.

Implementing setData allows users to test the implementation of the
component instead of the contract and is generally a bad practice (which is
why Adria, Lachlan, and I don’t use it or promote it ever)

**Comment by jackmu95** [comment]
You are right, we don't use it that much.
A PR would be really appreciated :)

**Comment by lmiller1990** [comment]
I think we *do* need to implement this - backwards compat is pretty important, and I think this feature is very easy to implement. I also expect a lot of push-back if we were to deprecate this. I would recommend implementing it and referencing it in the API reference part of the docs, but not including it in any of our guides (I think we settled on this been unideal in most cases - just use the `data` mounting option).

What does everyone think?

**Comment by jackmu95** [comment]
@lmiller1990 I would prefer to have this in v2 too.

A guide in the docs on why you should avoid using it and what better alternatives are seems to be a good idea.

**Comment by lmiller1990** [comment]
The reason it's not ideal is because arbitrarily setting data on a component is not something a user can do - ideally, test like a user does. Eg, if the data is set via a button, click the button with `trigger` etc. 

Either way, we should support this. I will make a PR in the near future unless someone else does first - you can reference this PR to see how it can be implemented.

**Comment by JessicaSachs** [comment]
Surprised to hear your stance on this -- I remember this being an unknown
amount of complexity. I'm happy to give users a non-breaking path. +1

**Comment by lmiller1990** [comment]
In terms of implementation, this was working https://github.com/vuejs/vue-test-utils-next/pull/52/files (this PR). Unless I am missing some specific use case?

Don't get me wrong, I still think `setData` sucks an is an anti-pattern... but this conflicts with "unopinionated utilities for testing". We can be opinionated in the docs 👍

**Comment by afontcu** [comment]
I'm also OK with adding `setData` for the sake of avoiding breaking changes, and document it as one of the "Advanced topics" (re https://github.com/vuejs/vue-test-utils-next-docs/issues/17#issue-611372985) VTU-next offers 👍

**Pull request closed without merging**